astropy/utils/data.py

```diff
@@ -1370,3 +1370,27 @@ def _release_download_cache_lock():
         msg = 'Error releasing lock. "{0}" either does not exist or is not ' +\
               'a directory.'
         raise RuntimeError(msg.format(lockdir))
+
+
+def _urls_on_shelf():
```

Comment:
And related to my comment in the line just below: if it is public, maybe the name should be `get_cached_urls`?